sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalSimplificationCodeFixProvider.cs

```diff
@@ -37,239 +37,250 @@ public sealed class ConditionalSimplificationCodeFixProvider : SonarCodeFixProvi
-                    SyntaxKind.CoalesceExpression,
-                    compared,
-                    whenFalse)
-                    .WithAdditionalAnnotations(annotation);
-                return createdExpression;
+                return CoalesceExpression(context.Compared, whenFalse, context.Annotation);
             }
-
-            if (CSharpEquivalenceChecker.AreEquivalent(whenFalse, compared))
+            else if (CSharpEquivalenceChecker.AreEquivalent(whenFalse, context.Compared))
             {
-                var createdExpression = SyntaxFactory.BinaryExpression(
-                    SyntaxKind.CoalesceExpression,
-                    compared,
-                    whenTrue)
-                    .WithAdditionalAnnotations(annotation);
-                return createdExpression;
+                return CoalesceExpression(context.Compared, whenTrue, context.Annotation);
             }
-
-            return GetSimplificationFromInvocations(whenTrue, whenFalse, null, compared, semanticModel, annotation,
-                isNullCoalescing: true);
+            return SimplifyInvocationExpression(context, whenTrue, whenFalse, null, isCoalescing: true);
         }
 
-        private static ExpressionSyntax GetSimplificationFromInvocations(ExpressionSyntax expression1, ExpressionSyntax expression2,
-            ExpressionSyntax condition, ExpressionSyntax compared, SemanticModel semanticModel, SyntaxAnnotation annotation,
-            bool isNullCoalescing)
+        private static ExpressionSyntax CoalesceAssignmentExpression(ComparedContext context, ExpressionSyntax assignmentExpression) =>
+            assignmentExpression is AssignmentExpressionSyntax originalAssignment
+                ? CoalesceAssignmentExpression(context, originalAssignment.Left, originalAssignment.Right)
+                : null;
+
+        private static ExpressionSyntax CoalesceAssignmentExpression(ComparedContext context, ExpressionSyntax left, ExpressionSyntax right)
         {
-            var methodCall2 = expression2 as InvocationExpressionSyntax;
-            if (!(expression1 is InvocationExpressionSyntax methodCall1) ||
-                methodCall2 == null)
+            if (context.IsCoalesceAssignmentSupported)
             {
-                return null;
+                try
+                {
+                    // Generating SyntaxFactory.BinaryExpression with SyntaxKindEx.CoalesceAssignmentExpression does not work with
+                    // currenlty referenced Roslyn version. Post-update using .WithOperatorToken() is also checked against unexpected arguments.
+                    var tree = CSharpSyntaxTree.ParseText("void F(object x) {x ??= null;}");
+                    var assignment = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>().Single();
+                    return assignment.Update(left, assignment.OperatorToken, right).WithAdditionalAnnotations(context.Annotation);
```

Comment:
I was curious as to what would go wrong here, as the `SyntaxKindEx.CoalesceAssignmentExpression` has the same value as `SyntaxKind.CoalesceAssignmentExpression` when we are using a recent roslyn version. 

I didn't get any issue here when remplacing the logic with`SyntaxFactory.AssignmentExpression(SyntaxKindEx.CoalesceAssignmentExpression, left, right).WithAdditionalAnnotations(context.Annotation);` So either we can remove this parsing mechanism here, or there are test cases missing that make this logic compulsory

Reply:
It seems that I was missusing `SyntaxFactory.BinaryExpression` instead of the correct `SyntaxFactory.AssignmentExpression` :-/